### ISSUE TITLE
feat: REST API source plugin

### ISSUE THREAD

**Issue by hermanschaaf** [cloudquery/cloudquery#9758]
### Which problem is this feature request solving?

It would be useful to have a generic REST API source plugin that can take an OpenAPI spec and fetch resources from the corresponding API. This could save days of development of tailored plugins or allow for syncing from custom sources, as long as they have an API (or can be wrapped with FastAPI).

### Describe the solution you'd like

The source config would probably take a path to an API spec, a base domain to use, and then use a standardized method to derive table names from endpoints. This is just one idea, we'll have to explore the area a bit to see what is possible.

### Pull request (optional)

- [ ] I can submit a pull request

**Comment by dbro** [comment]
@candiduslynx Please reconsider if this capability is worth investing in for CloudQuery. As an example, the [Airbyte Connector Builder](https://docs.airbyte.com/connector-development/connector-builder-ui/overview) which was [announced in May 2023](https://airbyte.com/blog/launching-the-no-code-connector-builder-build-custom-connectors-in-minutes) was very helpful to me to read from REST API sources. (However, Airbyte's performance was so slow and resource intensive that I am now looking for a better alternative.) The experience automates the schema spec creation, and offers convenient configuration settings for parent-child relationships (partitioning), auth, pagination, incremental sync, and basic field transformations.

**Comment by hermanschaaf** [comment]
I agree with @dbro, let's re-open and consider some more ways of introducing this. Even if it's only part of our [scaffold](https://github.com/cloudquery/cloudquery/tree/main/scaffold) tool, it could still be very useful.
